include/gc/Dialect/CPURuntime/IR/CPURuntimeOps.td

```diff
@@ -29,5 +28,57 @@ def CPURuntime_PrintfOp : CPURuntime_Op<"printf", [MemoryEffects<[MemWrite]>]>,
   }];
 }
 
+def CPURuntime_AllocOp : CPURuntime_Op<"alloc", [
+    AttrSizedOperandSegments,
+    MemoryEffects<[MemAlloc<DefaultResource>]>
+  ]> {
+
+  let summary = "CPURuntime memory allocation operation in a stack way.";
+  let description = [{
+    The `cpuruntime.alloc` operation allocates a region of memory on the CPU. It is
+    similar to the `std.alloc` op.
+
+    Example:
+
+    ```mlir
+    // For dynamic size memref
+    %memref = cpuruntime.alloc (%width) : memref<64x?xf32>
+    
+    // For static size memref
+    %memref = cpuruntime.alloc () : memref<64x32xf32>
+    
+    // For memref with symbolic operand in affine map
+    %memref = cpuruntime.alloc [%symbol_value] (%width) : memref<64x?xf32, affine_map<(d0, d1)[s0] -> (d0 + s0, d1)>
+
+    // For thread local alloc
+    %memref = cpuruntime.alloc threadLocal () : memref<64x32xf32>
+    ```
+  }];
+
+  let arguments = (ins Variadic<Index>:$dynamicSizes, Variadic<Index>:$symbolOperands, UnitAttr:$threadLocal);
+  let results = (outs Res<AnyMemRef, "", [MemAlloc<DefaultResource, 0, FullEffect>]>:$memref);
+
+  let assemblyFormat = [{
+    (`threadLocal` $threadLocal^)? `(` $dynamicSizes `)` (`` `[` $symbolOperands^ `]`)? attr-dict `:` type($memref)
+  }];
+}
+
+def CPURuntime_DeallocOp : CPURuntime_Op<"dealloc", [
+    MemoryEffects<[MemFree]>
+  ]> {
+
+  let summary = "CPU memory deallocation operation";
+
+  let description = [{
+    The `cpuruntime.dealloc` operation frees the region of memory referenced by a
+    memref which was originally created by the `cpuruntime.alloc` operation.
+    It is similar to the `std.dealloc` op.
```

Comment:
memref.dealloc?

Reply:
updated description, and add `thread_local` to dealloc as well. I think you are right, there's no need to attach `pooltype` in runtime. Keep the memorypool.cpp the same as before.